keep-ui/app/alerts/alert-sidebar.tsx

```diff
@@ -90,12 +99,21 @@ const AlertSidebar = ({ isOpen, toggle, alert }: AlertSidebarProps) => {
                   </div>
                 </Card>
                 <Card className="flex-grow">
-                  <AlertTimeline key={auditData? auditData.length : 1} alert={alert} auditData={auditData} isLoading={isLoading} onRefresh={handleRefresh} />
-                </Card>
-                <Card>
-                  <Title>Related Services</Title>
-                  <GraphVisualization demoMode={true} />
+                  <AlertTimeline
+                    key={auditData ? auditData.length : 1}
+                    alert={alert}
+                    auditData={auditData}
+                    isLoading={isLoading}
+                    onRefresh={handleRefresh}
+                  />
                 </Card>
+                <Title>Related Services</Title>
+                <TopologyPage
+                  providerId={alert.providerId || "NoProviderId"}
```

Comment:
why do you use those values and not just null?

Reply:
tbh for bad reasons, backend endpoint only filters topology when `providerId`, `environment` and `service` are given and so when they do not exist for some `alert`, I still want it to display empty state and not get all the topology. anyway, I'll make it cleaner